.github/workflows/pr-test.yml

```diff
@@ -57,15 +44,21 @@ jobs:
         if: always()
             
       - name: Archive test reports
-        uses: actions/upload-artifact@v1
+        uses: actions/upload-artifact@v3
         with:
           name: test-reports
           path: /tmp/test-reports
         if: always()
      
       - name: Publish Unit Test Results
-        uses: docker://ghcr.io/enricomi/publish-unit-test-result-action:v2
+        uses: EnricoMi/publish-unit-test-result-action@v2
         if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name == github.repository
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}
           files: ./**/build/test-results/**/*.xml
+
+  dependency-check:
```

Comment:
this should be part of `pr-build.yaml`

Reply:
Followed this to add depencycheck scan: https://github.com/hypertrace/java-grpc-utils/pull/46 and here its part of `pr-test.yml` 😅

Reply:
Two competing concerns here. The test job and build job have different triggers based on how they use secrets. We want to put all jobs under the `pull_request` trigger rather than `pull_request_target` unless there's a reason not to, so we'll need to take a look at how we can best arrange these so that we can get everything needed scheduled, but not over-expose on the the pull_request_target trigger. Will put something up later this morning.

Reply:
OK updated attribute service as an example - we don't really need that separation of pull_request and pull_request_target any more, so we can merge the two into a single workflow that gets scheduled with a few changes. You may want to do that in a separate PR though, up to you.

https://github.com/hypertrace/attribute-service/pull/170/files